### ISSUE TITLE
Team trolls

### ISSUE THREAD

**Issue by ghost** [A3Wasteland/ArmA3_Wasteland.Altis#468]
It seems team killer trolls have a new game recently, they repeatedly put down team mates so they are unconscious then revive them. Could you add a check/flag into far_revive please?

**Comment by AgentRev** [comment]
what kind?

**Comment by ghost** [comment]
I guess the option to punish a player if they are on the same team perhaps?

**Comment by AgentRev** [comment]
What about people who try to get run over on purpose so they can punish players and steal their ride?

**Comment by ghost** [comment]
Hmm good point :-(

**Comment by codexGW** [comment]
One thing that might keep the overall teamkill trolls at bay, is since teamkills are logged, maybe put a setting in the main config to perm ban to ind when over the set amount. A lot of the people I find who get reported for doing stuff like this, usually have a high teamkill number. It might also help farming for money a bit on op4 and blu4.

**Comment by AgentRev** [comment]
I mean, I could also put a punish option on revive menu, which would count as like 0.5 teamkill, although not in the near term.

Another option brought up in the past involved a TK counter that would decrease over time. Sorta like PlanetSide 2, where your weapons are locked if you inflict too much friendly damage within a certain period.

**Comment by WhatsUpDoc127** [comment]
Why not spectate them and if they keep doing what they are doing ban them for a week or so... JMHO

**Comment by hobart666** [comment]
My solution was to go independent only and use Revs group limiting to 6
players per group. But not everyone likes indie only. I'm forever low pop
so I don't mind.

**Comment by mawendt** [comment]
Why do anything? Point of wasteland is to remain wary of who you team up with, or associate with.  Plus - accountability in the Wasteland? Really? If you really want that, just log in to a coop server...

It becomes a learning experience to not group up with untrusted people.

Probably better if you simply hide non-group members from the map, even if they're on your team, and let nature take its course.

**Comment by hobart666** [comment]
>Why do anything? Point of wasteland is to remain wary of who you team up with, or associate with. Plus - accountability in the Wasteland? Really? If you really want that, just log in to a coop server...
>
>It becomes a learning experience to not group up with untrusted people.
>
>Probably better if you simply hide non-group members from the map, even if they're on your team, and let nature take its course.

This is precisely why I went indie only.

**Comment by ghost** [comment]
'Why do anything?' Awesome. 

With all due respects the servers you run are low pop so yeh jumping on and spectating newbies is much easier. You can't physically watch players 24/7 not when you've had over 200k. 

All I'm thinking about are the players and what can put them off from playing wasteland hence why I'm asking the main man if anything could be scripted to help.

**Comment by hobart666** [comment]
I meant to delete that part of his quote :P hence all the edits I did to
keep it from being too one sided. And to keep things like you're describing
down to 0 instances. But yessir you have TONS of players so I wouldn't
expect you or anyone else to follow my non lead.

**Comment by Soloknight** [comment]
Servers owner should be able to protect their player base, and these runts severely hurt it. I say the button in the revive dialog with a point system would be great.

**Comment by codexGW** [comment]
What I mentioned above will be pretty easy. Just add a check to the player load and add a option in the main config. Lodac did this almost 2 years ago. I might take a crack at it with the logging already done, if I get the time and will to do it. Here's the commits Lodac did for TOP Arma.
https://github.com/TOPARMA/ArmA3_Wasteland.Altis/commit/3bedc55ce9b940ae0c21c8357155df1c8d42737a
https://github.com/TOPARMA/ArmA3_Wasteland.Altis/commit/1ad33f5e11d325c55f24977e1a9cd510f9b4e052

**Comment by Soloknight** [comment]
The problem is the ones that actually get revived and don't have a option to punish.

**Comment by codexGW** [comment]
Agreed. I'm just thinking something that would take a little bit of time to implement. It might be good to have anyways. My vote goes towards the 0.5 team kill in revive with a eventual ban to ind. Maybe even add a timeout for the ban too.

**Comment by mawendt** [comment]
Luckily there are at least two 'punishment' scripts out there so there shouldn't be a whole lot of effort to add it. Though I'd recommend it as an admin optional select.

I have a different mindset, in that the popularity of Wasteland is not in the easiness or noob protection or whatever, but that it remain a challenge both against the game design as well as against human players. Including members of your own team. So if that is the philosophy,then no additional layers of protection or help is needed.

Over the years my unofficial interaction with players has always been a scavange, build, kill loop, with success mostly built into grouping up. The environment should maintain a set of rules that promotes that: you can have more fun as a free-for-all with a small unit; that building and defending your loot proves you are a seasoned player; that accomplishing missions before your competition gives satisfaction,particularly when you are under fire; that your ability to kill more than die earns you respect for cunning and smarts. 

Anything layered to limit that (like team killing protection) interrupts that - it takes the effort a player has to put into 'trust' away from him and moves it to game mechanics.

Yet if you want it, you can add it or have it as an option, like anything else but someone who is TK'g probably doesn't give a shit about anything but a ban. So jump to the end: if you TK, the offended party has an option to ban 3 days.

**Comment by AgentRev** [comment]
mawendt, the issue at stake here is players who are repeatedly teamkilling and reviving people just to anger them. It is completely retarded and these fucking baboons need to be dealt with, without admin intervention.

If it was possible to disable team damage, I would probably add an option, but damage handlers are rather limited in terms of player identification, so at best it would be buggy.

**Comment by WhatsUpDoc127** [comment]
This reminded me whenever Battlefield 2 was like this, people would kill somebody, just to the medic revive points. There are always new players who always try to ruin a game.

**Comment by mawendt** [comment]
Rev - you can still implement with the forgive scheme, and edit it for the admin of the server to allow for automatic bans. Most of it has already been done for you in similar scripting.

Again, as far as angering peeps - I understand that. I'm just putting out the opinion that creating a game restriction/environment rule (such as a forgive/kick or ban script) is probably not the best way of handling it. I was on a server that automatically threw the offending player into independent when he killed someone and didn't get 'forgiveness'. Seemed to work: one incident, unforgiven, and you are team blocked for 24 hrs to indie.

It would appear to me the best way of handling it would be to let players determine not to trust these completely fucktarded (is that a word? it should be) baboons and they become enemies. 

I'm not sure that creating a Protect-The-Snowflake script promotes any fair play. The idiots also team up and have one TeamSpeak ally on blue, one on opfor, and romp around allowing a TS buddy on indie make the kills or raid the bases.  Besides which, if they're killing their teammates, they care not about any kind of points or reputation counter on the server.

The observation is you really can't stop fucktards from being fucktards, so drive the environment back to players deciding who to trust and don't be bothered by the griefers. And if you don't have a banning request mechanism (as some servers do), it really leaves the best decision to drop bluefor/opfor and make everyone indie.

Not bragging, I have probably played over 1500 hrs of the various flavors of wasteland on a random servers (I'd have to check my steam, but I almost exclusively play Wasteland), and maybe a half dozen times I've been annoyed by players on the opfor team (I usually opfor it). So is it really a critical issue? Betcha no. But you may play on or have a particular server that could be managed a bit more... aggressively... by banning people with bad habits.

Hobart has the path. Though I'd bust up the team limit to ten.

**Comment by AgentRev** [comment]
Look, you've came here a number of times asking me to make vanilla A3W harder and indie-only. I'm gonna be loud and clear; that's not gonna happen under my watch. There's a truckload of shitty DayZ clones out there specifically tailored for that niche.

I'm a busy guy, say I wanna play a couple hours a week; I don't have time to deal with the trust crap, I just wanna blow shit up. That's why I don't play DayZ, Epoch, Exile and the likes. I either join a side to quickly find a group to latch onto, or I go indie lonewolf. It's that simple.

> if they're killing their teammates, they care not about any kind of points or reputation counter on the server

Exactly, if they don't care about it then they'll bust the fuse at some point. We just need the right kinda fuse. It's not meant to be some sort of meaningful reputation system, it's a disciplinary measure, nothing more.

**Comment by WhatsUpDoc127** [comment]
AgentRev just a suggestion but couldn't this guy who is complaining, be able to delete the Blufor and Opfor players that are in the map editor? Then whenever a player enters the server they can only join indie? I've seen a few servers that have blufor and indie only and maybe one or two that has indie only. 

If so I don't know why they come on here to complain whenever there are many things they can do themselves or at least try to learn BIS scripting. We have the ability to change things up on our own as server admins or like me the person who does everything in the editor and scripting side of things. Just my humble opinion. AgentRev gave us a server side mode that was left for dead by others and like I've said before AgentRev has given us a working mod and its up to US to do whatever we want with it from there. Look at the folks who created Wasteland on custom maps. They took the first step to do that. People should learn how to take that first step and learn the scripting. All the answers are on google search, trust me, if you have a question trust me somebody has already asked that question. This just my humble opinion.

**Comment by AgentRev** [comment]
The thing is, mawendt is not a server owner, he's a player. He wants me to change the mission to fit his tastes as to find more servers with such settings. I understand it might be frustrating for him not being able to find more hardcore servers, but trying to impose it on everyone else is not okay.

**Comment by WhatsUpDoc127** [comment]
Gotcha.....
